src/v2/view-builder/utils/DeviceFingerprinting.js

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright (c) 2021-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { $ } from 'okta';
+
+function getUserAgent() {
+  return navigator.userAgent;
+}
+
+function isWindowsPhone(userAgent) {
+  return userAgent.match(/windows phone|iemobile|wpdesktop/i);
+}
+
+export default {
+  isMessageFromCorrectSource($iframe, event) {
+    return event.source === $iframe[0].contentWindow;
+  },
+
+  // NOTE: This utility is similar to the DeviceFingerprint.js file used for V1 authentication flows.
+  //       Changes have been applied to this file ONLY to support OIE API enhancements.
+  generateDeviceFingerprint(fingerprintData) {
+    return new Promise((resolve, reject) => {
+      const userAgent = getUserAgent();
+      if (!userAgent) {
+        return reject('user agent is not defined');
+      } else if (isWindowsPhone(userAgent)) {
+        return reject('device fingerprint is not supported on Windows phones');
+      }
+
+      let $iframe;
+      let iFrameTimeout;
+      const self = this; // Needed for mocking
+
+      function removeIframe() {
+        $iframe.off();
+        $iframe.remove();
+        window.removeEventListener('message', onMessageReceivedFromOkta, false);
+      }
+
+      function handleError(reason) {
+        removeIframe();
+        reject(reason);
+      }
+
+      function onMessageReceivedFromOkta(event) {
+        if (!self.isMessageFromCorrectSource($iframe, event)) {
+          return;
+        }
+        // deviceFingerprint service is available, clear timeout
+        clearTimeout(iFrameTimeout);
+        if (!event || !event.data || event.origin !== fingerprintData.oktaDomainUrl) {
+          handleError('no data');
+          return;
+        }
+        try {
+          const message = JSON.parse(event.data);
+
+          if (message && message.type === 'FingerprintServiceReady') {
+            sendMessageToOkta({
+              type: 'GetFingerprint',
+              stateHandle: fingerprintData.stateHandle,
```

Comment:
Is there a PR on the server service to support `stateHandle`? 

Question 2: is this the only change from v1 file?

Reply:
There is! You can see how we plan to support this [in #51397](https://github.com/okta/okta-core/pull/51397).

And yes, this new `DeviceFingerprinting` logic is embedded within `v2` and will **only** be triggered for OIE flows. There will be no impact to legacy fingerprinting Widget-side.

Reply:
Sounds good. I meant to ask if this line is the only difference from the file used in v1, just to know exactly what are the differences.